### PR TITLE
Optional uncontrolled SearchLayout

### PR DESCRIPTION
**Pull request by stellarhoof**
Now we can do

```
<SearchLayout>
   .....
</SearchLayout>
```

and stuff will magically work. There's a small breaking change though. To control the search layout now you do:

```
<SearchLayout mode={...} setMode={...}>
    <Filters />
</SearchLayout>
```

instead of

```
<SearchLayout mode={...}>
    <Filters mode={...} setMode={...} />
</SearchLayout>
```

### CODE REVIEW

**Comment by decrapifier** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>The changelog has not been updated. Please update the changelog.</td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>The version was not updated. Please update the version.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>The README has not been updated. Please update the README.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>Could not find any browser results.</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>We were able to automatically fix some formatting issues in this PR for you!</td>
    </tr>
  </tbody>
</table>


### Some things that were possibly fixed:

- Code that could be fixed via the --fix flag
- Formatting that could be fixed by prettier

Take a look at this commit to see what happened in detail: 28a4e6347ef8b01a4efef34b69284e5e6ea2ed50


And look at this wiki page to see the reasoning behind the ESLint rules: https://github.com/smartprocure/eslint-config-smartprocure/wiki/Rules-and-Why-We-Chose-Them
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against ff20ff92d4cb242ef66ef0807a2cc302f55d4871
</p>

**Comment by stellarhoof** [comment]
@dubiousdavid @michsa @Akrion Putting you guys as reviewers as well since you're probably interested from the discussion we had yesterday on standup.

File: src/themes/greyVest/SearchLayout.js [hunk 1]
```diff
@@ -1,13 +1,24 @@
...
 
-let SearchLayout = ({ mode, ...props }) => (
-  <div className={`gv-search-layout-${mode}`} {...props} />
-)
+let SearchLayoutContext = React.createContext()
+
+export let useSearchLayout = () => useContext(SearchLayoutContext)
+
+let SearchLayout = ({ mode = 'basic', ...props }) => {
+  let layout = useLocalStore(() => ({ mode }))
```
Comment (daedalus28) [inline src/themes/greyVest/SearchLayout.js]: This is a breaking change and makes `mode` into a modeDefault

Comment (stellarhoof) [reply src/themes/greyVest/SearchLayout.js]: See my comments above on breaking changes

**Comment by stellarhoof** [comment]
> It seems like we should still support a non-context version in case we need to modify mode from somewhere else (like page state).

See comment in the description for how to modify layout state using `useSearchLayout`. The question is whether we should go with the breaking change or not.

**Comment by stellarhoof** [comment]
Blocked on https://github.com/smartprocure/contexture-react/pull/259

File: src/SearchLayout.js [hunk 2]
```diff
@@ -11,9 +17,15 @@ let styles = mode => ({
     mode === 'basic' ? 'minmax(250px, 400px) minmax(0, 1fr)' : 'minmax(0, 1fr)',
 })
 
-let SearchLayout = ({ mode, style, ...props }) => (
-  <div style={{ ...styles(mode), ...style }} {...props} />
-)
+let SearchLayout = ({ style, mode, setMode, ...props }) => {
+  let [stateLayout] = React.useState(observable.box('basic'))
+  let layout = mode ? F.stateLens([mode, setMode]) : stateLayout
```
Comment (michsa) [inline src/SearchLayout.js]: What's the observable for? I suspect I'm missing something, but this one-liner seems to work just as well:

```es6
let layout = F.stateLens(mode ? [mode, setMode] : React.useState('basic'))
```

Comment (stellarhoof) [reply src/SearchLayout.js]: Hmm yeah you're right we don't need the observable. Good catch :+1: 

FYI: Using hooks conditionally like that is not recommended: https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

**Comment by Akrion** [comment]
@stellarhoof conflicts

**Comment by stellarhoof** [comment]
@Akrion Done

**Comment by stellarhoof** [comment]
This is stale

**Pull request closed without merging**